Server/Controllers/TransactionController.cs

```diff
@@ -1,81 +1,49 @@
+        private readonly IMediator mediator;
 
-        public TransactionController(ApplicationDbContext context, UserManager<ApplicationUser> userManager)
+        public TransactionController(UserManager<ApplicationUser> userManager, IMediator mediator)
         {
-            this.context = context;
             this.userManager = userManager;
+            this.mediator = mediator;
         }
 
         [HttpGet]
         [Route("{typeFilter}/{sortFilter}/{orderFilter}/{itemsPerPage}/{pageNumber}")]
         public TransactionHistoryData GetTransactions(string typeFilter,string sortFilter,string orderFilter, int itemsPerPage, int pageNumber)
         {
-            var transactionData = new TransactionHistoryData { 
+            var transactionData = new TransactionHistoryData
+            {
                 TransactionsList = new List<TransactionDto>()
             };
-            var transactionsList = new List<Transaction>();
-            var userId = userManager.GetUserId(User);
-            var idList = context.Users.Include(e => e.Wallets).FirstOrDefault(x => x.Id == userId).Wallets.Select(e => e.Id).ToList();
 
-            transactionsList = typeFilter switch
+            var query = new GetTransactionsQuery
             {
-                "Made" => context.Transactions.Where(t => idList.Contains(t.SourceWalletId) && t.DestinationUserId != t.SourceUserId).ToList(), //for outgoing transactions
-                "Recived" => context.Transactions.Where(t => idList.Contains(t.DestinationWalletId) && t.DestinationUserId != t.SourceUserId).ToList(), //for recived transactions
-                "Intern" => context.Transactions.Where(t => idList.Contains(t.SourceWalletId) && idList.Contains(t.DestinationWalletId)).Distinct().ToList(), //for transactions betweem our own wallets
-                _ => context.Transactions.Where(t => idList.Contains(t.SourceWalletId) || idList.Contains(t.DestinationWalletId)).Distinct().ToList(), //for all transactions
+                UserId = userManager.GetUserId(User),
+                TypeFilter = typeFilter,
+                SortFilter = sortFilter,
+                OrderFilter = orderFilter
             };
 
-            transactionsList = sortFilter switch
-            {
-                "Date" => transactionsList = orderFilter switch
-                {
-                    "Ascendent" => transactionsList.OrderBy(e => e.Date).ToList(),
-                    _ => transactionsList.OrderByDescending(e => e.Date).ToList()
-                },
-
-
-                "Currency" => transactionsList = orderFilter switch
-                {
-                    "Ascendent" => transactionsList.OrderBy(e => e.Currency).ThenBy(e => e.Amount).ToList(),
-                    _ => transactionsList.OrderByDescending(e => e.Currency).ThenByDescending(e => e.Amount).ToList()
-                },
+            var transactionsList = mediator.Send(query).GetAwaiter().GetResult();
```

Comment:
It's generally not a good idea to Wait for a task.
You could simply make this controller action async. Why not?

Reply:
I changed it . Thx